src/test_l2_dsCompositeIn.c

```diff
@@ -1,119 +1,214 @@
+*
+* **Test Procedure:**
+* Refer to UT specification documentation [dsCompositeIn_L2_Low-Level_TestSpecification.md](../../docs/pages/dsCompositeIn_L2_Low-Level_TestSpecification.md)
+*/
+
+void test_l2_dsCompositeIn_VerifyCompositeInputStatus(void)
+{
+    gTestID = 2;
+    UT_LOG_INFO("In %s [%02d%03d]\n", __FUNCTION__, gTestGroup, gTestID);
 
-#include <string.h>
-#include <stdlib.h>
+    dsError_t ret;
+    dsCompositeInStatus_t status;
 
-#include <ut.h>
-#include <ut_log.h>
+    UT_LOG_DEBUG("Invoking dsCompositeInInit()");
+    ret = dsCompositeInInit();
+    UT_ASSERT_EQUAL_FATAL(ret, dsERR_NONE);
+
+    UT_LOG_DEBUG("Invoking dsCompositeInGetStatus()");
+    ret = dsCompositeInGetStatus(&status);
+    UT_ASSERT_EQUAL(ret, dsERR_NONE);
+    if (ret != dsERR_NONE)
+    {
+        UT_LOG_ERROR("dsCompositeInGetStatus failed with error: %d\n", ret);
+        dsCompositeInTerm(); // Call post-requisite API on failure
+        return;
+    }
+
+    UT_ASSERT_EQUAL(status.isPresented, false);
+    UT_ASSERT_EQUAL(status.isPortConnected[dsCOMPOSITE_IN_PORT_0], false);
+    UT_ASSERT_EQUAL(status.isPortConnected[dsCOMPOSITE_IN_PORT_1], false);
+    UT_ASSERT_EQUAL(status.activePort, dsCOMPOSITE_IN_PORT_NONE);
+
+    UT_LOG_DEBUG("Invoking dsCompositeInTerm()");
+    ret = dsCompositeInTerm();
+    UT_ASSERT_EQUAL_FATAL(ret, dsERR_NONE);
+
+    UT_LOG_INFO("Out %s\n", __FUNCTION__);
+}
 
 /**
-* @brief TODO: Describe the object of the test
+* @brief Test to verify the selection and status of Composite In Port
 *
-* TODO: Add the description of what is tested and why in this test
+* This test verifies the functionality of Composite In Port selection and its status. It checks if the port is initialized, selected, and terminated correctly. It also checks the status of the port after selection.
 *
-* **Test Group ID:** TODO: Add the group this test belongs to - Basic (for L1): 01 / Module (L2): 02 / Stress (L2): 03)@n
-* **Test Case ID:** TODO: Add the ID of the test case so that it can be logically tracked in the logs@n
+* **Test Group ID:** 02@n
+* **Test Case ID:** 003@n
 *
 * **Test Procedure:**
-* TODO: Refer to L2 Specification documentation : [l2_module_test_specification_template.md](../docs/pages/l2_module_test_specification_template.md)
+* Refer to UT specification documentation [dsCompositeIn_L2_Low-Level_TestSpecification.md](../../docs/pages/dsCompositeIn_L2_Low-Level_TestSpecification.md)
 */
-void test_l2_dsCompositeIn (void)
+
+void test_l2_dsCompositeIn_VerifyCompositeInPortSelectionAndStatus(void)
 {
-	UT_FAIL("This function needs to be implemented!");
+    gTestID = 3;
+    UT_LOG_INFO("In %s [%02d%03d]\n", __FUNCTION__, gTestGroup, gTestID);
+
+    dsError_t ret;
+    dsCompositeInStatus_t status;
+
+    ret = dsCompositeInInit();
+    UT_LOG_DEBUG("Invoking dsCompositeInInit()");
+    UT_ASSERT_EQUAL_FATAL(ret, dsERR_NONE);
+
+    for (dsCompositeInPort_t port = dsCOMPOSITE_IN_PORT_0; port < dsCOMPOSITE_IN_PORT_MAX; port++)
```

Comment:
Should read the supported ports from the profile rather than considering the max supported?